libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/src/azure/core/az_base64.c

```diff
@@ -404,7 +403,7 @@ AZ_NODISCARD az_result
 az_base64_decode(az_span destination_bytes, az_span source_base64_text, int32_t* out_written)
 {
   _az_PRECONDITION_VALID_SPAN(destination_bytes, 1, false);
-  _az_PRECONDITION_VALID_SPAN(source_base64_text, 4, false);
```

Comment:
Please add all 4 examples of base64 text to validate all possibilities introduced by the code below.